src/preprocess/run_tg.py

```diff
@@ -68,17 +68,56 @@
                 '--out':  superpop_split.get_pfile_path(node=node, fold_index=fold_index, part_name=part_name)
                 }, args_list=['--make-pgen'])
 
-                # # Run PCA on train and save weights for projection
-                # if part_name == 'train':
-                #     run_plink(args_list=['--pfile', superpop_split.get_pfile_path(node=node, fold_index=fold_index, part_name=part_name),
-                #                          '--freq', 'counts',
-                #                          '--out', superpop_split.get_pca_path(node=node, fold_index=fold_index, part=part_name, ext=''),
-                #                          '--pca', 'allele-wts', '20'],
-                #               )
-                # # Use saved PCA weights for all projections
-                # run_plink(args_list=['--pfile', superpop_split.get_pfile_path(node=node, fold_index=fold_index, part_name=part_name),
-                #                      '--read-freq', superpop_split.get_pca_path(node=node, fold_index=fold_index, part='train', ext='.acount'),
-                #                      '--score', superpop_split.get_pca_path(node=node, fold_index=fold_index, part='train', ext='.eigenvec.allele'), '2', '5', 'header-read', 'no-mean-imputation', 'variance-standardize', '--score-col-nums', '6-25',
-                #                      '--out', superpop_split.get_pca_path(node=node, fold_index=fold_index, part=part_name, ext='')]
-                #           )
+    for fold_index in range(10):
+        # Perform centralized sample ids merge to use it with `--keep` flag in plink
+        ids = []
+
+        for node in set(TG_SUPERPOP_DICT.values()):
+            ids_filepath = superpop_split.get_ids_path(
+                fold_index=fold_index,
+                part_name='train',
+                node=node
+            )
+
+            with open(ids_filepath, 'r') as ids_file:
```

Comment:
why are you reading and writing like this and not with pd.read_csv(fn, sep='\t') and .to_csv ?